code/datums/components/religious_tool.dm

```diff
@@ -0,0 +1,149 @@
+/**
+  *
+  * Allows the parent to act similarly to the Altar of Gods with modularity. Invoke and Sect Selection is done via attacking with a bible. This means you cannot sacrifice Bibles (you shouldn't want to do this anyways although now that I mentioned it you probably will want to).
+  *
+  */
+/datum/component/religious_tool
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+	/// Enables access to the global sect directly
+	var/datum/religion_sect/easy_access_sect
+	/// What extent do we want this religious tool to act? In case you don't want full access to the list. Generated on New
+	var/operation_flags
+	/// The rite currently being invoked
+	var/datum/religion_rites/performing_rite
+	///Sets the type for catalyst
+	var/catalyst_type = /obj/item/storage/book/bible
+	///Enables overide of COMPONENT_NO_AFTERATTACK, not recommended as it means you can potentially cause damage to the item using the catalyst.
+	var/force_catalyst_afterattack = FALSE
+
+/datum/component/religious_tool/Initialize(_flags = ALL, _force_catalyst_afterattack = FALSE, override_catalyst_type)
+	. = ..()
+	SetGlobalToLocal() //attempt to connect on start in case one already exists!
+	operation_flags = _flags
+	force_catalyst_afterattack = _force_catalyst_afterattack
+	if(override_catalyst_type)
+		catalyst_type = override_catalyst_type
+
+/datum/component/religious_tool/RegisterWithParent()
+	RegisterSignal(parent, COMSIG_RELIGION_NEWSECT, .proc/SetGlobalToLocal)
+	RegisterSignal(parent, COMSIG_PARENT_ATTACKBY,.proc/AttemptActions)
+	RegisterSignal(parent, COMSIG_PARENT_EXAMINE, .proc/on_examine)
+
+/datum/component/religious_tool/UnregisterFromParent()
+	UnregisterSignal(parent, list(COMSIG_PARENT_ATTACKBY,COMSIG_RELIGION_NEWSECT, COMSIG_PARENT_EXAMINE))
+
+/**
+  * Sets the easy access variable to the global if it exists.
+  */
+/datum/component/religious_tool/proc/SetGlobalToLocal()
+	if(!GLOB.religious_sect)
+		return
+	easy_access_sect = GLOB.religious_sect
+
+
+/**
+  * Since all of these involve attackby, we require mega proc. Handles Invocation, Sacrificing, And Selection of Sects.
+  */
+/datum/component/religious_tool/proc/AttemptActions(datum/source, obj/item/the_item, mob/living/user)
+	/**********Sect Selection**********/
+	if(!easy_access_sect)
+		if(!operation_flags & RELIGION_TOOL_SECTSELECT)
+			return
+		. = COMPONENT_NO_AFTERATTACK //At this point you're intentionally trying to select a sect.
```

Comment:
it'd still be this if these were in separate procs because the main proc is so filled up :^(

Reply:
breathe